src/MerchantCenter/PriceBenchmarks.php

```diff
@@ -21,6 +22,196 @@
+					'benchmark_price_currency_code' => $price_competitiveness['benchmark_price_currency_code'] ?? '',
+					'benchmark_price_micros'        => round( $benchmark_price_micros / 1000000, 2 ),
+					'price_gap'                     => round( $price_gap / 1000000, 2 ),
+					'suggested_price'               => isset( $price_insights['suggested_price_micros'] )
+						? round( $price_insights['suggested_price_micros'] / 1000000, 2 )
+						: '',
+					'suggested_price_currency_code' => $price_insights['suggested_price_currency_code'] ?? '',
+					'predicted_impressions_change'  => $price_insights['predicted_impressions_change_fraction'] ?? '',
+					'predicted_clicks_change'       => $price_insights['predicted_clicks_change_fraction'] ?? '',
+					'predicted_conversions_change'  => $price_insights['predicted_conversions_change_fraction'] ?? '',
+				];
+			},
+			$mapped_data
+		);
+
+		return $response_data;
+	}
+
+	/**
+	 * Gets the product ID from the Google product ID.
+	 *
+	 * @param string $mc_product_id Simple product ID (`merchant_center_id`) or
+	 *                              namespaced product ID (`online:en:GB:merchant_center_id`)
+	 * @return int the ID for the WC product linked to the provided Google product ID (0 if not found)
+	 */
+	public function get_product_id( string $mc_product_id ): int {
+		// Maybe remove everything before the last colon ':'
+		$mc_product_id_tokens = explode( ':', $mc_product_id );
+		$mc_product_id        = end( $mc_product_id_tokens );
+
+		// Support a fully numeric ID both with and without the `gla_` prefix.
+		$wc_product_id = 0;
+		$pattern       = '/^(' . preg_quote( $this->get_slug(), '/' ) . '_)?(\d+)$/';
+		$wc_pattern    = '/^(woocommerce_gpf_)?(\d+)$/';
+		if ( preg_match( $pattern, $mc_product_id, $matches ) ) {
+			$wc_product_id = (int) $matches[2];
+		} elseif ( preg_match( $wc_pattern, $mc_product_id, $matches ) ) {
+			$wc_product_id = (int) $matches[2];
+		}
+
+		/**
+		 * This filter is documented in ProductHelper::get_wc_product_id.
+		 *
+		 * @see ProductHelper::get_wc_product_id
+		 */
+		return (int) apply_filters( 'woocommerce_gla_get_wc_product_id', $wc_product_id, $mc_product_id );
+	}
+
+	/**
+	 * Retrieves the product thumbnail URL.
+	 *
+	 * @param int $product_id WooCommerce product ID.
+	 * @return string|null Product thumbnail URL or null if not found.
+	 */
+	protected function get_product_thumbnail( int $product_id ): ?string {
+		$thumbnail_id = get_post_thumbnail_id( $product_id );
+
+		if ( ! $thumbnail_id ) {
+			return '';
+		}
+
+		$thumbnail_url = wp_get_attachment_url( $thumbnail_id );
+
+		return $thumbnail_url ?? '';
+	}
+
+	/**
+	 * Retrieves performance data (clicks and conversions) for a given product ID.
+	 *
+	 * @param int|string $id The product ID to fetch performance data for.
+	 * @return array Associative array with 'clicks' and 'conversions' keys, or empty array on failure.
+	 */
+	public function get_performance_data( $id ): array {
```

Comment:
This method shouldn't be needed if we can call `$merchant->get_merchant_performance_data()` directly, similar to how we are calling

```
$benchmark_data = $merchant->get_price_comparisons_data( $args );
$price_insights_data = $merchant->get_price_insights_data( $args );
```